internal/crypto/cert.go

```diff
@@ -109,6 +177,40 @@ func (c *TLSCertificateConfig) GetPEMBytes() ([]byte, []byte, error) {
 	return certBytes, keyBytes, nil
 }
 
+func (ca *CA) IssueRequestedClientCertificateAsX509(csr *x509.CertificateRequest, expirySeconds int) (*x509.Certificate, error) {
+	return ca.IssueRequestedCertificateAsX509(csr, expirySeconds, []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth})
```

Comment:
Why does this need to be wrapped?

Reply:
Otherwise we have to supply key use constants in the upper layers. F.E. - we need to modify the certificatesigningrequest.go in the service package to pass key usage.

I am happy either way. I did not want to touch those without necessity and I wanted to keep the changes mostly to crypto/

Reply:
What if we just used the `internal_ca`'s `IssueRequestedCertificateAsX509` directly, and some other function to translate it to bytes as necessary? I think it would be easier to follow and there could be fewer functions overall. Whatever is calling this `IssueRequestedClientCertificateAsX509` could just call `ca.IssueRequestedCertificateAsX509` with these parameters.

I see in certificatesigningrequest we do call it [here](https://github.com/flightctl/flightctl/blob/main/internal/service/certificatesigningrequest.go#L92). I think in this case it will be fine to change that line to the new function name and parameters.